### PR TITLE
Drop support for python 3.6, transition to PEP-621 pyproject.toml, and misc project updates

### PR DESCRIPTION
**Pull request by Secrus**
Decided to put a few minutes into updating the project. 
- dropped Python 3.6, since it's no longer supported,
- added support for Python 3.9 and 3.10,
- added common virtualenv directory names to `.gitignore`,
- updated tools URLs and versions in `.pre-commit-config.yaml`,
- migrated project metadata to PEP-621 spec in `pyproject.toml`,
- moved `.coveragerc` data to `pyproject.toml` (and removed unnecessary entries)

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def lint(session):
     session.run("pre-commit", "run", "--all-files", *session.posargs)
 
 
-@nox.session(python=["3.6", "3.7", "3.8"])
+@nox.session(python=["3.7", "3.8", "3.9", "3.10"])
```
Comment (consideRatio) [inline noxfile.py:28]:
```suggestion
@nox.session(python=["3.7", "3.8", "3.9", "3.10", "3.11"])
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python: [3.6, 3.7, 3.8]
+        python: [3.7, 3.8, 3.9, "3.10"]
```
Comment (consideRatio) [inline .github/workflows/ci.yml:20]:
```suggestion
        python: ["3.7", "3.8", "3.9", "3.10", "3.11"]
```

**Review by consideRatio — APPROVED** [review]
Looks great to me, nice!

**Comment by Secrus** [comment]
@consideRatio I have refreshed the PR (Python 3.11 and updated tools versions). Please re-review.

**Review by consideRatio — APPROVED** [review]
Looks good to me!

**Review by choldgraf — APPROVED** [review]
Looks good to me - thanks for this improvement! Happy to merge once tests are passing - it seems there are some failures in there.

cc @Secrus in case you're not watching this one.

**Comment by Secrus** [comment]
I will look at fixing this within the next few days.

**Comment by consideRatio** [comment]
Observed failures

```
nox > Command flit install -s --deps=production --extras test failed with exit code 1:
Unexpected names under [project]: home-page         W-flit_core.config
license field should be <class 'dict'>, not <class 'str'>
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,37 +1,56 @@
...
   "Intended Audience :: Developers",
   "License :: OSI Approved :: MIT License",
   "Operating System :: OS Independent",
   "Programming Language :: Python",
   "Programming Language :: Python :: 3",
-  "Programming Language :: Python :: 3.6",
   "Programming Language :: Python :: 3.7",
   "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
```
Comment (consideRatio) [inline pyproject.toml:21]:
```suggestion
  "Programming Language :: Python :: 3.10",
  "Programming Language :: Python :: 3.11",
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,37 +1,56 @@
...
 home-page = "https://github.com/executablebooks/sphinx-autobuild"
 license = "MIT"
-requires-python = ">=3.6"
-requires = [
+requires-python = ">=3.7"
+dependencies = [
   "sphinx",
   "livereload",
   "colorama",
 ]
```
Comment (consideRatio) [inline pyproject.toml:32]:
```suggestion

[project.urls]
Documentation = "https://github.com/executablebooks/sphinx-autobuild#readme"
Issues = "https://github.com/executablebooks/sphinx-autobuild/issues"
Source = "https://github.com/executablebooks/sphinx-autobuild"

```

File: pyproject.toml [hunk 5]
```diff
@@ -1,37 +1,56 @@
...
   "Programming Language :: Python :: 3",
-  "Programming Language :: Python :: 3.6",
   "Programming Language :: Python :: 3.7",
   "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
   "Topic :: Software Development :: Libraries :: Python Modules",
 ]
 home-page = "https://github.com/executablebooks/sphinx-autobuild"
 license = "MIT"
```
Comment (consideRatio) [inline pyproject.toml:25]:
```suggestion
license = {file = "LICENSE"}
```

File: pyproject.toml [hunk 6]
```diff
@@ -1,37 +1,62 @@
...
   "Programming Language :: Python :: 3",
-  "Programming Language :: Python :: 3.6",
   "Programming Language :: Python :: 3.7",
   "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
   "Topic :: Software Development :: Libraries :: Python Modules",
 ]
 home-page = "https://github.com/executablebooks/sphinx-autobuild"
```
Comment (consideRatio) [inline pyproject.toml:25]:
```suggestion
```

**Comment by consideRatio** [comment]
@Secrus I became too curious and ended up resolving this. Can you review my added commits, and if they look good to you I figure we go for a merge!

0da788a - Supplemental fixes to pyproject.toml
87cf689 - ci: update versions of github workflow's actions
10de017 - pre-commit: fix config of black

**Comment by Secrus** [comment]
@consideRatio LGTM, thanks!

**Pull request merged**